content/scripts/build.js

```diff
@@ -1610,6 +1612,69 @@ class Builder {
           }
         });
 
+        if (this.options.fixFlaws) {
+          // For flaws that can be changed, change the source itself.
+
+          // Copy of the original raw HTML that we're going to (potentially)
+          // repeatedly do string replaces on.
+          let newRawHtml = rawHtml;
+          flaws
+            .filter((flaw) => {
+              // Only interested in 'MacroRedirectedLinkError' flaws because
+              // them we can do something about.
+              // The flaw might be from a prerequisite, if so, leave it alone
+              // See https://github.com/mdn/yari/issues/772
+              return (
+                flaw.name === "MacroRedirectedLinkError" &&
+                (!flaw.filepath || path.dirname(flaw.filepath) === folder)
+              );
+            })
+            .forEach((flaw, i) => {
+              if (!rawHtml.includes(flaw.macroSource)) {
+                throw new Error(
+                  `rawHtml doesn't contain macrosSource (${flaw.macroSource})`
+                );
+              }
+              const newMacroSource = flaw.macroSource.replace(
+                flaw.redirectInfo.current,
+                flaw.redirectInfo.suggested
+              );
+              newRawHtml = newRawHtml.replace(flaw.macroSource, newMacroSource);
+
+              // If the flaw could be fixed, it's no longer a flaw. So remove
+              // it from the array.
+              // This means you could run the CLI like this:
+              // `build ... --fix-flaws --flaw-levels="macros:error"`
+              // it would only complain about this that couldn't be fixed.
+              if (newRawHtml !== rawHtml) {
```

Comment:
I thought a LOT about this and logic was wrong. But your fix wasn't right. The reason is that we will always change `newRawHtml` on every `newRawHtml.replace(..., ...)`. For example;
```javascript
var text = "peter"

var flaws = ['e', 'b'];
flaws.forEach((flaw, i) => {
  if (!text.includes(flaw)) {
    throw new Error(`${flaw} not in original text!`);
  }
  text = text.replace(flaw, 'X');  // HERE!!
  flaws.splice(i, 1);
})
console.log(flaws)
console.log(text)
```
that `text.replace(flaw, 'X')` will always change the string. 
The output:
```
[ 'b' ]
pXter
```

And once we fix https://github.com/mdn/yari/issues/756#issuecomment-654313496 the new code will be equivalent to:

```diff
var text = "peter"

-var flaws = ['e', 'b'];
+var flaws = ['e', 'b', 'e'];
flaws.forEach((flaw, i) => {
  if (!text.includes(flaw)) {
    throw new Error(`${flaw} not in original text!`);
  }
  text = text.replace(flaw, 'X');  // HERE!!
  flaws.splice(i, 1);
})
console.log(flaws)
console.log(text)
```
and it will still work, but now the output will be:

```
[ 'b' ]
pXtXr
```